internal/gatewayapi/address.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"net/netip"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+var _ AddressesTranslator = (*Translator)(nil)
+
+type AddressesTranslator interface {
+	ProcessAddresses(gateways []*GatewayContext, xdsIR XdsIRMap, infraIR InfraIRMap, resources *Resources)
+}
+
+func (t *Translator) ProcessAddresses(gateways []*GatewayContext, xdsIR XdsIRMap, infraIR InfraIRMap, resources *Resources) {
+	for _, gateway := range gateways {
+		// Infra IR already exist
+		irKey := irStringKey(gateway.Gateway)
+		gwInfraIR := infraIR[irKey]
+
+		ipAddr := sets.Set[string]{}
+
+		for _, addr := range gateway.Spec.Addresses {
+			switch *addr.Type {
+			case v1beta1.IPAddressType:
+				if _, err := netip.ParseAddr(addr.Value); err == nil {
+					ipAddr.Insert(addr.Value)
```

Comment:
can we expose the error in the status condition ?

Reply:
actually, we cannot expose this error in the status condition.

because the address will be assigned to `Service.ExternalIPs` before we reach here:
https://github.com/envoyproxy/gateway/blob/f81c0118ca0c487db16ee3d54ae21e94a705a348/internal/status/gateway.go#L25

so instead set error status condition ourselves, we will directly get error from k8s.io.api, result in failure creating the Gateway Service.

Reply:
actually, we cannot expose this error in the status condition.

becasue when we apply an invalid IP address (like `a.b.c.d`) to Gateway,  it will fail to create a Gateway Service, 

https://github.com/envoyproxy/gateway/blob/f81c0118ca0c487db16ee3d54ae21e94a705a348/internal/infrastructure/kubernetes/service.go#L20-L32

thus we cannot get this Service when setting the status condition:

https://github.com/envoyproxy/gateway/blob/f81c0118ca0c487db16ee3d54ae21e94a705a348/internal/status/gateway.go#L25-L28

IMO, we can solve this by validating the `Gateway.Spec.Addresses` before creating the Service, since the GW API does not provide any validation webhook on this field, should we validate it ourselves or raise an issue in upstream?

cc @arkodg

Reply:
thanks for raising an issue upstream !